### PR TITLE
Fix some options of MNIST example not working

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

File: examples/mnist/train_mnist_custom_loop.py [hunk 1]
```diff
@@ -51,6 +51,10 @@ def main():
     optimizer = chainer.optimizers.Adam()
     optimizer.setup(model)
 
+    if args.resume:
+        # Resume from a snapshot
+        serializers.load_npz(args.resume, model)
```
Comment (beam2d) [inline examples/mnist/train_mnist_custom_loop.py]: Should `optimizer` be loaded as well?

**Comment by hvy** [comment]
@kmaehashi thanks for this fix. I assume that `args.resume` should be the directory corresponding to `args.out`. File names should be appended in the code when loading the snapshots (`mlp.model`, `mlp.state`).

**Comment by hvy** [comment]
Ping @kmaehashi

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
LGTM, @beam2d could you approve?

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
@kmaehashi I think this fix should be backported. What do you think?

**Comment by kmaehashi** [comment]
@hvy I agree to backport it.

**Comment by hvy** [comment]
@kmaehashi Thanks for the label, I created a backport.

**Pull request merged**